atlasdb-api/src/main/java/com/palantir/atlasdb/cell/api/DdlManager.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cell.api;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import java.util.Map;
+
+public interface DdlManager {
```

Comment:
I want ONLY user data tables to go through this. Internal tables should have their own initialization mechanism. I don't want the internal migration service to be notified of the internal tables for now, to contain risk. 

So this opinion may change, but for risk mitigation, I really want the new codepaths to only be hit with user data tables. Hence the like *Legacy methods in Schema, because, as pointed out in internal chat, Sweep etc. goes through this codepath and tries to re-use the basic "get all tables from a Schema and create them" logic, even if it doesn't actually end up using half of the actual features this provides (like TableMetadata I think really doesn't make sense for these tables, and yet we end up storing this stuff in all the KVSes.

I think the nice benefit and the way where this should go is that once we add a #getTables method here, it should ONLY return user data tables. Which simplifies a lot of other code that currently needs to know about internal atlas tables and like do some weird filtering.

Reply:
Alright, yep makes sense (given the package it is specifically for user APIs).

We do actually customise that (you might notice oddly suspicious things like compression chunk length kbs or some other weird booleans in the table metadata that _are_ set for transactions2, but not for any other table)! Though we would be able to unravel that

Reply:
Ah right, I see now.